### ISSUE TITLE
Refactor code for changing managing entity for a group when updating the head of household 

### ISSUE THREAD

**Issue by f-odhiambo** [opensrp/fhircore#2940]
Resolve issues with persistent Household Head on change managing entity for family.

This is a follow-up ticket 

Screen Recording 
[Screencast from 14-12-2023 10:54:20 ASUBUHI.webm](https://github.com/ona-health/sid-dto-fhir-resources/assets/79141719/15a4528c-3042-4511-87a3-eb4d57449240)

**Comment by f-odhiambo** [comment]
@f-odhiambo had a paring troubleshooting session with @ellykits and Identified some of the potential methods that need to be changed, will share an implementation plan once the triage is complete

**Comment by f-odhiambo** [comment]
This issue is completed, and merged via PR #2973. Needs documetation as a follow up

**Comment by ellykits** [comment]
merged.
